server/sonar-server/src/test/java/org/sonar/server/user/ws/SetHomepageActionTest.java

```diff
@@ -19,20 +19,50 @@
  */
 package org.sonar.server.user.ws;
 
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 import org.sonar.api.server.ws.WebService;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.DbTester;
+import org.sonar.db.component.ComponentDbTester;
+import org.sonar.db.component.ComponentDto;
+import org.sonar.db.component.ComponentTesting;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.db.user.UserDto;
+import org.sonar.server.exceptions.UnauthorizedException;
+import org.sonar.server.tester.UserSessionRule;
+import org.sonar.server.ws.TestResponse;
 import org.sonar.server.ws.WsActionTester;
 
+import static org.apache.http.HttpStatus.SC_NO_CONTENT;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.sonar.server.user.ws.SetHomepageAction.PARAM_TYPE;
+import static org.sonar.server.user.ws.SetHomepageAction.PARAM_VALUE;
+import static org.sonarqube.ws.Users.CurrentWsResponse.HomepageType.MY_ISSUES;
+import static org.sonarqube.ws.Users.CurrentWsResponse.HomepageType.MY_PROJECTS;
+import static org.sonarqube.ws.Users.CurrentWsResponse.HomepageType.ORGANIZATION;
+import static org.sonarqube.ws.Users.CurrentWsResponse.HomepageType.PROJECT;
 
 public class SetHomepageActionTest {
 
-  private SetHomepageAction underTest = new SetHomepageAction();
-  private WsActionTester wsTester = new WsActionTester(underTest);
+  @Rule
+  public UserSessionRule session = UserSessionRule.standalone();
```

Comment:
rename to `userSession`. We can have dbSession for example, so it can be misleading

Reply:
done